sources/mongodb/helpers.py

```diff
@@ -0,0 +1,113 @@
+"""SQL database source helpers"""
+
+import json
+from typing import (
+    Any,
+    Optional,
+    Iterator,
+)
+
+import dlt
+from dlt.sources.credentials import ConnectionStringCredentials
+from dlt.common.configuration.specs import BaseConfiguration, configspec
+from dlt.common.typing import TDataItem
+
+from pymongo import MongoClient
+from pymongo.collection import Collection
+from pymongo.cursor import Cursor
+from datetime import datetime
+
+
+class CollectionLoader:
+    def __init__(
+        self,
+        client: MongoClient,
+        collection: Collection,
+        incremental: Optional[dlt.sources.incremental[Any]] = None,
+    ) -> None:
+        self.client = client
+        self.collection = collection
+        self.incremental = incremental
+        if incremental:
+            self.cursor_field = incremental.cursor_path
+            self.last_value = incremental.last_value
+        else:
+            self.cursor_column = None
+            self.last_value = None
+
+    def make_query(self) -> Cursor[Any]:
+        if not self.incremental or not self.last_value:
+            filter_op = {}
+        elif (
+            self.incremental.last_value_func is max
+        ):  # Filtered according to last_value function
+            filter_op = {self.cursor_field: {'$gte': self.last_value}}
+        elif self.incremental.last_value_func is min:
+            filter_op = {self.cursor_field: {'$lt': self.last_value}}
+        else:  # Custom last_value, load everything and let incremental handle filtering
+            filter_op = {}
+        cursor = self.collection.find(filter_op)
+        return cursor
+    
+    def clean_document(self, document: dict) -> dict:
+        """Clean document to be json serializable"""
+        output = {}
+        if not isinstance(document, dict):
+            return document
+        for key, value in document.items():
+            if isinstance(value, dict):
+                output[key] = self.clean_document(value)
+            elif isinstance(value, list):
+                output[key] = [self.clean_document(item) for item in value]
+            elif isinstance(value, datetime):
+                output[key] = value
+            else:
+                output[key] = str(value)
```

Comment:
hmmm why? we are losing types here. dlt will create schema from common python types so they should stay (including bytes if mongo supports binary blobs).
some problems with bson having their own types?